File: 0338-counting-bits/0338-counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int>ans;
        for(int i=0;i<=n;i++){
            int x=i,c=0;
            for(int j=0;x>0;j++){
                if(x%2==1)  c+=1;
                x/=2;
            }
            ans.push_back(c);
        }
        return ans;
    }
};

File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[n + 1];
        for (int i = 0; i <= n; i++) {
            int c = 0;
            int temp = i; 
            while (temp > 0) { 
                if ((temp & 1) == 1) 
                    c += 1;
                temp >>= 1; 
            }
            ans[i] = c;
        }
        return ans;
    }
}